launcher/src/backend/ServiceManager.js

```diff
@@ -177,14 +177,69 @@ export class ServiceManager {
         break;
 
       case "reSync":
-        //initiate resync
+        await this.resyncService(service, data);
         break;
 
       default:
         break;
     }
   }
 
+  async resyncService(serviceID, checkpointUrl) {
+    let services = await this.readServiceConfigurations();
+    let client = services.find((service) => service.id === serviceID);
+    const checkpointCommands = {
+      LighthouseBeaconService: "--checkpoint-sync-url=",
+      LodestarBeaconService: "--checkpointSyncUrl=",
+      PrysmBeaconService: "--checkpoint-sync-url=",
+      NimbusBeaconService: "--trusted-node-url=",
+      TekuBeaconService: "--initial-state=",
+    };
+    const dataDir = client.volumes.find(
+      (vol) =>
+        vol.servicePath === "/opt/app/beacon" ||
+        vol.servicePath === "/opt/app/data" ||
+        vol.servicePath === "/opt/data/geth" ||
+        vol.servicePath === "/opt/data/erigon"
+    ).destinationPath;
+    await this.manageServiceState(client.id, "stopped");
+
+    if (dataDir.length > 0) {
+      await this.deleteDataVolume(dataDir);
+    }
+    //Prysm specific
+    if (client.service == "PrysmBeaconService") {
```

Comment:
You should avoid client specific exceptions. You could have checked if `service.command` is a string. If so -> convert to array and later back to string. This way you can handle all clients the same way and dont need this big `if else block`.

The reason behind this is if there is another client thats `command` is a string, I'd have to add this one here too. Or if Prysm is refactored and has no longer `command` as a string then this doesn't work as well.